### ISSUE TITLE
Feature request: Allow automatic clear of output window before running task

### ISSUE THREAD

**Issue by BillDenton** [Microsoft/vscode#19510]
- VSCode Version: Code 1.8.1 (ee428b0eead68bf0fb99ab5fdc4439be227b6281, 2016-12-19T14:49:23.350Z)
- OS Version: Windows_NT ia32 6.1.7601
- Extensions:

|Extension|Author|Version|
|---|---|---|
|read-only-indicator|alefragnani|0.6.0|
|alignment|annsk|0.3.0|
|Doxygen|bbenoist|1.0.0|
|vscode-astyle|chiehyu|0.4.0|
|python|donjayamanne|0.5.6|
|EditorConfig|EditorConfig|0.3.4|
|copy-relative-path|enochc|0.1.1|
|matlab|Gimly81|0.4.0|
|status-bar-tasks|GuardRex|0.1.2|
|vscode-dimmer|hoovercj|2.1.0|
|docthis|joelday|0.3.10|
|cpptools|ms-vscode|0.10.0|
|copy-file-name|nemesv|1.0.0|
|vscode-clearcase|OpenNingia|1.4.2|
|vscode-icons|robertohuertasm|7.1.2|
|theme-cyanicpastels|SamuelLampa|0.0.9|
|indenticator|SirTori|0.2.1|
|vscode-hexdump|slevesque|1.1.4|
|code-spell-checker|streetsidesoftware|0.14.8|
|comment-snippets|Tobiah|0.2.0|
|dark-plus-material|vangware|0.1.4|
|change-case|wmaurer|1.0.0|
|highlight-trailing-white-spaces|ybaumes|0.0.2|

---

Steps to Reproduce:

1. Run a task.
2. Output is appended to output window.
Expected behaviour: Output window is cleared before starting the task so it easier to see the results of the current task. This should be a configurable option. Probably a global one plus a per task one.

**Comment by BillDenton** [comment]
@dbaeumer Any update on this?

**Comment by dbaeumer** [comment]
This got partly addressed with the new terminal runner which support a presentation property that lets you control the panel usages. For example

```ts
			"presentation": {
				"panel": "new"
			},
````

which creates a new panel for every run (but it doesn't automatically clear /close and old one). 

The story forward is to use the terminal runner not the old output runner. If you think there is something missing for you please open a new issue. I will close this one since no work is planned to support new features in the output runner.

**Comment by BillDenton** [comment]
How do we use the "terminal runner" and get the output cleared at the start of running a task?

**Comment by dbaeumer** [comment]
June release notes or https://code.visualstudio.com/docs/editor/tasks#_convert-from-010-to-200

**Comment by BillDenton** [comment]
I have already converted to "version": "2.0.0", but had not added "runner": "terminal". After adding this my existing tasks (or at least one of them) seem to be working
```
  "version": "2.0.0",
  "type": "shell",
  "presentation": {
    "echo": true,
    "reveal": "always",
    "focus": false,
    "panel": "dedicated"
  },
  "runner": "terminal",
  "tasks": [
    {
```
What new can I now do? 
I've just tried adding add "cls" to the start of my tasks, but it didn't work:
```
"args": [
        "cls &&",
        "my_build_program options"
]

```
BTW: I'm using an external batch file to beep at the end of the build, so I don't have to watch the long builds:
```
(my_build_prog || beepbeepbeep.bat)
&&
beep.bat
```
I get three beeps if it fails and one if it succeeds.

**Comment by dbaeumer** [comment]
The cls trick doesn't work. You get a fresh console process anyways, however in the UI by default we append the output unless you create a new panel on every run.

**Comment by BillDenton** [comment]
I don't want to create a new panel each run as I would end with a lot of panels to delete. Is there any way of clearing the panel (i.e the equivalent of Terminal Clear) when the build starts?

**Comment by dbaeumer** [comment]
No, but there is a feature request for it here: https://github.com/Microsoft/vscode/issues/30058
